contracts/test/CompoundGovernor.t.sol

```diff
@@ -736,3 +769,190 @@ contract CastVoteWithReasonAndParams is CompoundGovernorTest {
+        uint256 _snapShot = governor.proposalSnapshot(_proposalId);
+        assertEq(_snapShot, _clockAtSubmit + INITIAL_VOTING_DELAY);
+    }
+
+    function testFuzz_ZeroReturnedIf_ProposalSnapshotCalledWithInvalidProposalId(
+        uint256 _invalidProposalId
+    ) public view {
+        uint256 _snapShot = governor.proposalSnapshot(_invalidProposalId);
+        assertEq(_snapShot, 0);
+    }
+}
+
+contract ProposalEta is CompoundGovernorTest {
+    function testFuzz_ProposalEtaCorrectWithEnumeratedId(uint256 _proposerIndex) public {
+        _proposerIndex = bound(_proposerIndex, 0, _majorDelegates.length - 1);
+        address _proposer = _majorDelegates[_proposerIndex];
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = _submitProposal(_proposer, _proposal);
+        _passProposal(_proposalId);
+        vm.roll(vm.getBlockNumber() + INITIAL_VOTING_PERIOD + 1);
+        uint256 _timeOfQueue = block.timestamp;
+        governor.queue(_proposalId);
+        uint256 _eta = governor.proposalEta(_proposalId);
+        assertEq(_eta, _timeOfQueue + timelock.delay());
+    }
+
+    function testFuzz_ZeroReturnedIf_ProposalEtaCalledWithInvalidProposalId(
+        uint256 _invalidProposalId
+    ) public view {
+        uint256 _eta = governor.proposalEta(_invalidProposalId);
+        assertEq(_eta, 0);
+    }
+}
+
+contract ProposalProposer is CompoundGovernorTest {
+    function testFuzz_ProposalProposerCorrectWithEnumeratedId(uint256 _proposerIndex) public {
+        _proposerIndex = bound(_proposerIndex, 0, _majorDelegates.length - 1);
+        address _proposerExpected = _majorDelegates[_proposerIndex];
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = _submitProposal(_proposerExpected, _proposal);
+        address _proposer = governor.proposalProposer(_proposalId);
+        assertEq(_proposerExpected, _proposer);
+    }
+
+    function testFuzz_ZeroReturnedIf_ProposalProposerCalledWithInvalidProposalId(
+        uint256 _invalidProposalId
+    ) public view {
+        address _proposer = governor.proposalProposer(_invalidProposalId);
+        assertEq(_proposer, address(0));
+    }
+}
+
+contract ProposalNeedsQueing is CompoundGovernorTest {
+    function testFuzz_ProposalNeedsQueuingCorrectWithEnumeratedId(uint256 _proposerIndex) public {
+        _proposerIndex = bound(_proposerIndex, 0, _majorDelegates.length - 1);
+        address _proposerExpected = _majorDelegates[_proposerIndex];
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = _submitProposal(_proposerExpected, _proposal);
+        bool _queuingNeeded = governor.proposalNeedsQueuing(_proposalId);
+        assertEq(_queuingNeeded, true);
+    }
+}
+
+contract ProposalThreshold is CompoundGovernorTest {
+    function test_ProposalThreshold() public view {
+        assertEq(governor.proposalThreshold(), INITIAL_PROPOSAL_THRESHOLD);
+    }
+}
```

Comment:
~nit: might want to test that after it's set again, the getter reflects the new value

Reply:
Done: 6fbb5c589c622b9f9def544caf7ce183e15e525d